legate/settings.py

```diff
@@ -58,5 +64,134 @@ class LegateRuntimeSettings(Settings):
         """,
     )
 
+    test: EnvOnlySetting[bool] = EnvOnlySetting(
+        "test",
+        "LEGATE_TEST",
+        default=False,
+        convert=convert_bool,
+        help="""
+        Whether to enable test execution.
+
+        This is a read-only environment variable setting used by the runtime.
+        """,
+    )
+
+    min_gpu_chunk: EnvOnlySetting[int] = EnvOnlySetting(
+        "min_gpu_chunk",
+        "LEGATE_MIN_GPU_CHUNK",
+        default=1048576,  # 1 << 20
+        test_default=2,
+        convert=convert_int,
+        help="""
+        Minimum chunk size to enable GPU execution.
+
+        This is a read-only environment variable setting used by the runtime.
+        """,
+    )
+
+    min_cpu_chunk: EnvOnlySetting[int] = EnvOnlySetting(
+        "min_cpu_chunk",
+        "LEGATE_MIN_CPU_CHUNK",
+        default=16384,  # 1 << 14
+        test_default=2,
+        convert=convert_int,
+        help="""
+        Minimum chunk size to enable CPU execution.
+
+        This is a read-only environment variable setting used by the runtime.
+        """,
+    )
+
+    min_omp_chunk: EnvOnlySetting[int] = EnvOnlySetting(
+        "min_omp_chunk",
+        "LEGATE_MIN_OMP_CHUNK",
+        default=131072,  # 1 << 17
+        test_default=2,
+        convert=convert_int,
+        help="""
+        Minimum chunk size to enable CPU execution.
+
+        This is a read-only environment variable setting used by the runtime.
+        """,
+    )
+
+    window_size: EnvOnlySetting[int] = EnvOnlySetting(
+        "window_size",
+        "LEGATE_WINDOW_SIZE",
+        default=1,
+        test_default=1,
+        convert=convert_int,
+        help="""
+        Window size.
+
+        This is a read-only environment variable setting used by the runtime.
+        """,
+    )
+
+    max_pending_exceptions: EnvOnlySetting[int] = EnvOnlySetting(
+        "max_pending_exceptions",
+        "LEGATE_MAX_PENDING_EXCEPTIONS",
+        default=64,
+        test_default=1,
+        convert=convert_int,
+        help="""
+        Maximum number of pending exceptions.
+
+        This is a read-only environment variable setting used by the runtime.
+        """,
+    )
+
+    precise_exception_trace: EnvOnlySetting[bool] = EnvOnlySetting(
+        "precise_exception_trace",
+        "LEGATE_PRECISE_EXCEPTION_TRACE",
+        default=False,
+        test_default=False,
+        convert=convert_bool,
```

Comment:
Of note here, the C++ code will not accept all the "truthy" values that `convert_bool` will accept (i.e. "yes", "Yes", "on" etc.), it will only accept 0 and 1 (and other integers I suppose, anything non-zero will count as "true").

We should either add a note specifically for C++-only boolean options, that they need to be passed strictly with 0/1, or add an `extract_env_bool` function to the C++ side that accepts the same strings as `convert_bool`, and use it when a boolean envvar is expected.

Reply:
Yes, that's reasonable. I would suggest also adding a comment on the top of the "settings" page saying that boolean values are passed through the environment as 0/1.

Reply:
https://github.com/nv-legate/legate.core/pull/684/commits/4c4362708e84501450163f1c110274dd3f5c7ad2 and https://github.com/nv-legate/legate.core/pull/684/commits/6b8846ac642fc1bdcd18a8d82fdbca9426230fde made the comment show on every individual property:

![image](https://user-images.githubusercontent.com/1078448/232849398-f0d997db-82c9-4b37-9a06-28a2d18d7732.png)